__tests__/app.test.js

```diff
@@ -122,3 +122,112 @@ describe("GET /api/articles", () => {
+        const { comments } = body;
+        expect(comments).toBeInstanceOf(Array);
+        expect(comments).toHaveLength(11);
+        const commentsCopy = [...comments];
+        const sortedComments = commentsCopy.sort((commentA, commentB) => {
+          return commentA.created_at - commentB.created_at
+        });
+        expect(comments).toEqual(sortedComments);
+        sortedComments.forEach((comment) => {
+          expect(comment).toHaveProperty('comment_id', expect.any(Number));
+          expect(comment).toHaveProperty('body', expect.any(String));
+          expect(comment).toHaveProperty('votes', expect.any(Number));
+          expect(comment).toHaveProperty('author', expect.any(String));
+          expect(comment).toHaveProperty('article_id', expect.any(Number));
+          expect(comment).toHaveProperty('created_at')
+          expect(new Date(comment.created_at)).toEqual(expect.any(Date));
+          });
+      });
+  });
+  test("GET 204: responds with 'No comments found for this article!' for a valid article_id with no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(204)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe("No comments found for this article!");
+      });
+  });
+  test("GET 404: responds with not found if article_id has not been found", () => {
+    return request(app)
+      .get("/api/articles/100/comments")
+      .expect(404)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe("Requested information not found!");
+      });
+  });
+  test("GET 400: responds with 'Bad article_id!' when article id entered in wrong format", () => {
+    return request(app)
+      .get("/api/articles/wrong_format/comments")
+      .expect(400)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe('Bad article_id!');
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("POST 201: adds a new comment to the database and responds with posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments/")
+      .send({
+        body: 'New comment to post',
+        username: 'icellusedkars'
+      },)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.postedComment).toEqual({
+          comment_id: expect.any(Number),
+          body: 'New comment to post',
+          article_id: 1,
+          author: 'icellusedkars',
+          votes: 0,
+          created_at: body.postedComment.created_at
+        })
+      });
+  });
+  test("POST 400: responds with 'Bad Request! ...' if article_id has not been found", () => {
+    return request(app)
+      .post("/api/articles/500/comments")
+      .send({
+        body: 'New comment to post',
+        username: 'icellusedkars'
+      },)
+      .expect(400)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe('Bad Request! Key (article_id)=(500) is not present in table \"articles\".');
```

Comment:
I would be hesitant to give away too much information about how your database is working.